### PR TITLE
Add eyes test for Curriculum Catalog Card

### PR DESCRIPTION
**Pull request by megcrenshaw**
Adds eyes test for curriculum catalog. There are four UI scenarios that need to be added as well:
1) Signed-out user
2) Signed-in student
3) Signed-in teacher with sections, assigning both a Unit and UnitGroup to sections
4) Signed-in teacher without sections

The big question mark is how variable the curricula are on the test machine. I'm seeing some scripts being seeded at https://github.com/code-dot-org/code-dot-org/blob/868942d292a60d18a2db26c37efab7f5511015ef/dashboard/lib/tasks/seed.rake#L79-L140, but notably, "Jigsaw" is not in there, and Course Offerings do not necessarily correspond to seeded scripts. There is also this seed which explicitly seeds course offerings, but there are many more course offerings on our test machine than get seeded here.
https://github.com/code-dot-org/code-dot-org/blob/868942d292a60d18a2db26c37efab7f5511015ef/dashboard/lib/tasks/seed.rake#L303-L307

I propose we don't investigate further at this point, but if Jigsaw or other course offerings are constantly changing, we go back in and make the offerings on our machine more consistent.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/MultipleSectionsAssigner.jsx [hunk 1]
```diff
@@ -195,24 +195,13 @@ const MultipleSectionsAssigner = ({
 
   return (
     <AccessibleDialog onClose={onClose}>
-      <div
-        tabIndex="0"
-        className={classnames(
-          'uitest-confirm-assignment-dialog',
```
Comment (megcrenshaw) [inline apps/src/templates/MultipleSectionsAssigner.jsx:201]: @levadadenys noted this was in the file twice. When I looked at it, I couldn't see evidence it was being used (a different UI test uses this classname but with a different component)

File: dashboard/test/ui/features/acquisition_products/curriculum_catalog.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Then I click selector "span:contains(Digital Literacy)"
+    And I wait until element "h4:contains(AI for Oceans)" is not visible
+    And I see no difference for "Curriculum Catalog: One Offering"
+
+    Then I click selector "#grade-dropdown-button"
+    And I wait until element "span:contains(Grade 12)" is visible
+    Then I click selector "span:contains(Grade 12)"
+    And I wait until element "h5:contains(No matching curricula)" is visible
+    And I see no difference for "Curriculum Catalog: No Offerings"
+    And I close my eyes
```
Comment (TurnerRiley) [inline dashboard/test/ui/features/acquisition_products/curriculum_catalog.feature:22]: Really love this testing of going all the way to 'No offerings'!

**Review by TurnerRiley — APPROVED** [review]

File: dashboard/test/ui/features/acquisition_products/curriculum_catalog.feature (hunk 2 above)
Comment (megcrenshaw) [reply dashboard/test/ui/features/acquisition_products/curriculum_catalog.feature:22]: The sad penguin deserves some love!

File: apps/src/templates/MultipleSectionsAssigner.jsx (hunk 1 above)
Comment (levadadenys) [reply apps/src/templates/MultipleSectionsAssigner.jsx:201]: Awesome, thanks for deleting these two!

**Review by levadadenys — APPROVED** [review]

**Pull request merged**